url7606378bcc_ljy2010a_protobuf2_tgz-pJ8-com_game_gameObj_bufBuilder_ReflectBuilderJ8/src/com/game/gameObj/bufBuilder/ReflectBuilder.java

```diff
@@ -0,0 +1,108 @@
+package com.game.gameObj.bufBuilder;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import com.game.gameObj.Soldier;
+import com.game.protobuf.GameProto;
+import com.game.protobuf.ModelAttr.SoldierAttr;
+
+public class ReflectBuilder {
+
+    public final static String GameObjTypeField = "GameObjType";
+
+    public final static String GameObjIdField = "GameObjId";
+
+    public final static String ModelAttrPackage = "com.game.protobuf.ModelAttr$";
+
+    public static GameProto.GameAction getGameAction(Object arg_Origin, boolean isSyncCoor, String ActionType, Object... Attrs) throws NoSuchMethodException, SecurityException, IllegalAccessException, IllegalArgumentException, InvocationTargetException, ClassNotFoundException, NoSuchFieldException, InstantiationException {
+
+        GameProto.GameObject.Builder gameGameObjectBuilder = GameProto.GameObject.newBuilder();
+
+        Method getGameObjTypeMethod = arg_Origin.getClass().getMethod("get" + GameObjTypeField);
+
+        String GameObjType = (String) getGameObjTypeMethod.invoke(arg_Origin);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Catch `InvocationTargetException` explicitly when you call `Method.invoke()`. Use `Throwable.getCause()` or `getTargetException()` in the catch handler to access more information about the underlying exception. See the following diffs on GitHub that make this code change: [diff1](https://github.com/EsfingeFramework/gamification/commit/20792047e6c69457cca03e877f2ada578188f32a#diff-6808c12fe9ddaea042db84f95dd5ba78adab99d0cb18bb4d85916a8723530768L21), [diff2](https://github.com/cucumber-attic/cuke4duke/commit/f4821d400cf2590126f1298a8dff9f7a33d45dc5#diff-80adaffc20c0e6e7d2e42312386ffe1ddc10213bf5b8e8919895fd85d666c8deL34)

Learn more: [InvocationTargetException in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/InvocationTargetException.html)